app/quest/completed/page.tsx

```diff
@@ -0,0 +1,102 @@
+"use client";
+
+import React, { useContext } from "react";
+import { useRouter } from "next/navigation";
+import Button from "@components/UI/button";
+import styles from "@styles/quests.module.css";
+import moduleStyles from "@styles/category.module.css";
+import homeStyles from "@styles/Home.module.css";
+import { QuestsContext } from "@context/QuestsProvider";
+import { pickRandomObjectsFn } from "@utils/quest";
+import buttonStyles from "@styles/components/button.module.css";
+import Lottie from "lottie-react";
+import verifiedLottie from "@public/visuals/verifiedLottie.json";
+import CategoryTitle from "@components/UI/titles/categoryTitle";
+import Quest from "@components/quests/quest";
+
+export default function Page() {
+  const router = useRouter();
+  const { completedQuestIds, categories } = useContext(QuestsContext);
+
+  const computeIncompleteQuests = categories.map((item) => {
```

Comment:
can you put this variable in a state ? putting simple variabled might cause the variable to recompute again and again on each rerender. I think you can put this in a `useMemo` to avoid this recompute

Reply:
Done